specs/lazy-adr/adr-011-based-sequencing.md

```diff
@@ -0,0 +1,133 @@
+# ADR 011: Based Sequencing
+
+## Changelog
+
+- 2025-03-11: Initial draft
+
+## Context
+
+This ADR provides details regarding based sequencing using Celestia as base layer, Rollkit as rollup consensus layer, and EVM as execution layer.
+
+### Based Sequencing
+
+User transactions are ordered by the base layer.
+
+### Rollkit
+
+Is a combination of three layers:
+
+* Sequencing layer
+* Consensus layer
+* Execution layer
+
+#### Sequencing layer
+
+* Implements sequencing interface
+* Popular implementations are centralized sequencing and based sequencing
+
+#### Consensus layer
+
+* Two types of nodes: Proposer (aka header producer) and full node.
+* Proposer provides a soft-commitment (or state preconfs) by preparing the batch (according to the global forkchoice rule that is acceptable by all nodes in the rollup network, e.g., full nodes), executing it, producing the update state root, and proposing a header with updated state root and proof data for verification for later hard-confirmation. Same for both centralized and based sequencing.
+* Full node receives transaction batches and headers via p2p gossip (state preconf from the proposer). Using the proof data in the header, full node can perform full verification to hard confirm.
+
+#### Execution layer
+
+* Implements the execution interface
+* Popular implementations are evm and abci
+* Mempool and RPC are part of the execution layer, hence if a user submits a rollup transaction, it gets to the mempool on the execution layer, which then pulled by the consensus layer (by calling GetTxs) to submit to sequencing layer, which finally submits the transaction to based layer.
```

Comment:
could we remove the rollkit section, we can assume the reader has knowledge of it. Could we put some more info on what is based sequencing and possibly link to some examples

Reply:
adding the based sequencing context, but would like to keep the rollkit section for self-sufficiency.